Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orientation.h

```diff
@@ -36,22 +35,27 @@ namespace CGAL {
 namespace Polygon_mesh_processing {
 
 namespace internal{
-  template <class Less_xyz, class VPmap>
-  struct Compare_vertex_points_xyz_3{
-    Less_xyz less;
+
+  template <class GT, class VPmap>
+  struct Compare_vertex_points_z_3
+  {
+    typename GT::Compare_z_3 compare_z;
     VPmap vpmap;
 
-   Compare_vertex_points_xyz_3(VPmap const& vpmap)
-	: vpmap(vpmap){}
+    Compare_vertex_points_z_3(VPmap const& vpmap, const GT& gt)
+      : vpmap(vpmap)
+    {
+      compare_z = gt.compare_z_3_object();
+    }
```

Comment:
The following works also if `GT::Compare_z_3` does not have a default constructor (may happen with a geom traits with state):
```c++
Compare_vertex_points_z_3(VPmap const& vpmap, const GT& gt)
     : compare_z(gt.compare_z_3_object()), vpmap(vpmap)
{}
```